### PR TITLE
DCOS-40871: use InfoBox component to render banners and alerts

### PR DESCRIPTION
**Pull request by mperrotti**
## Testing
Just confirm these updates didn't break any of the design. This design element is used all over the UI, but here are some places to check:

**Create service modal**
1. Tap the "+" icon in the "Services" tab
2. Add a new single-container service
3. Click "Review & Run" without giving the service an ID
4. The red error message that appears at the top of the form should look pretty much the same as it did before these changes 

**Package detail page**
1. Click a package in the "Packages" tab
2. The yellow box with the pre-install notes should look pretty much the same as it did before these changes

**Framework configuration**
1. Click a package in the "Packages" tab
2. Click "Review & Run" and complete the flow
3. Repeat steps 1 and 2 with the same package
4. The pre-install notes and the error that tells you you can't install the same package twice should look pretty much the same as they did before these changes

**Jobs form**
1. Go to the "Jobs" tab and create a new job
2. Click the "Create job" without giving your job an id
3. The red error banner should look pretty much the same as it did before these changes

This is not a complete list of everywhere this component is being used, but I think it's enough coverage to confirm nothing is broken

## Trade-offs
None that I can think of

## Dependencies
There's a branch that's also named `mp/chore/DCOS-40871-use-infobox` in dcos-ui-plugins-private. This uses `InfoBoxBanner` in place of the deleted `Banner` component.

## Screenshots
Probably not worth it for this PR, but I'm happy to add some if reviewers would find that helpful

### CODE REVIEW

**Comment by nLight** [comment]
Legit system tests fails. I bet we just have to update the way we look for the messages.
https://github.com/dcos/dcos-ui/blob/dima/cypress-dot-separated-input/system-tests/universe/test-universe.js#L45
https://github.com/dcos/dcos-ui/blob/dima/cypress-dot-separated-input/system-tests/services/test-apps.js#L96

```
ERROR 2018-10-26 15:29:20 system:  ✘ Services Applications fails create the same app name again
ERROR 2018-10-26 15:29:20 system:  ✘ Universe fails to install a package with the same name
```

**Comment by leemunroe** [comment]
While testing this PR I noticed a lot of spacing issues. The headings and paragraphs seem to be missing margins. 

Before (CCM)
image

After (this PR)
image

**Comment by leemunroe** [comment]
I think this message might have been overlooked — it doesn't seem to be using InfoBox.
image

**Comment by mperrotti** [comment]
Hey @leemunroe, here's the PR mentioned above: https://github.com/dcos-labs/ui-kit/pull/244/files

I tested it in dcos-ui, and this is the result:
screen shot 2018-10-26 at 7 48 16 pm

**Comment by DanielMSchmidt** [comment]
I added a blocked label as we need the PR in the UI kit first

**Comment by GeorgiSTodorov** [comment]
Create service modal, Framework configuration and Jobs form all look different from before.

### Before
screenshot from 2018-11-06 12-05-52

### After
screenshot from 2018-11-06 12-05-45


### Before
screenshot from 2018-11-06 12-07-32

### After
screenshot from 2018-11-06 12-07-39


### Before
screenshot from 2018-11-06 12-08-22

### After
screenshot from 2018-11-06 12-08-13

**Comment by nLight** [comment]
@mperrotti it needs a manual rebase 👐

File: plugins/services/src/js/service-configuration/PodContainersConfigSection.js [hunk 1]
```diff
@@ -26,10 +26,11 @@ const PodContainersConfigSection = ({ appConfig, onEditClick }) => {
     return (
       <div>
         <ConfigurationMapHeading level={2}>Containers</ConfigurationMapHeading>
-        <Alert>
-          No containers specified! Please specify at least one container when
-          creating a multi-container definition!
-        </Alert>
+        <InfoBoxInline
+          appearance="danger"
+          message="No containers specified! Please specify at least one container when
```
[the remark this answers is not in the document]
Comment (mperrotti) [reply plugins/services/src/js/service-configuration/PodContainersConfigSection.js]: Yes, some people actually prefer it 🤷‍♂️ . In this case, Prettier did it automatically

**Comment by GeorgiSTodorov** [comment]
Failing tests and snapshots.

**Comment by GeorgiSTodorov** [comment]
Snapshots can be easily fixed by running `npm run test -- -u`. In the tests, it looks like it can't find the banner. I looked into it a bit, but I am not sure how to fix it.

screenshot from 2018-11-08 10-30-50

**Comment by mperrotti** [comment]
Tests run fine for me locally without any snapshots being updated. Also, this branch was passing in CI last week 🤔

**Comment by GeorgiSTodorov** [comment]
CI is broken right now.

**Comment by mperrotti** [comment]
Ah ok, I thought it was just being problematic for integration tests

**Pull request merged**